storybookjs/react-native
# feat: enables use of docgen for generated arg types (somewhat experimental)

Issue: 

## What I did

provides addArgTypesEnhancer as an export so a user could setup arg type generation.

## How to test

Please explain how to test your changes and consider the following questions

- run example app and see that boolean and text input examples have args generated.

## How a user can set it up

install `@storybook/react`, `@storybook/docs-tools` and `babel-plugin-react-docgen-typescript`

add the following plugin to your babel:
`plugins: [['babel-plugin-react-docgen-typescript', { exclude: 'node_modules' }]],`

and add this code before getStorybookUI

```
import { extractArgTypes } from '@storybook/react/dist/modern/client/docs/extractArgTypes';
import { addArgTypesEnhancer, addParameters } from '@storybook/react-native';
import { enhanceArgTypes } from '@storybook/docs-tools';

addArgTypesEnhancer(enhanceArgTypes);
addParameters({
  docs: {
    extractArgTypes,
  },
});

```

In the example I add it to a file called doctools and import it in Storybook.tsx
